### ISSUE TITLE
Breaking to labeled block outside inline for loop crashes compiler

### ISSUE THREAD

**Issue by Hejsil** [ziglang/zig#1609]
```zig
test "" {
    failed: {
        // We have to iterate twice
        inline for ([]void{{},{}}) |_| {
            // a, has to be runtime known
            var a = false;

            // We have to have this condition + break twice
            if (a)
                break :failed;
            if (a)
                break :failed;
        }
    }
}
```

```
zig/src/list.hpp:37: T ZigList<T>::pop() [with T = IrInstruction*]: Assertion `length >= 1' failed.
[1]    13661 abort (core dumped)  zig1 test test2.zig
```

**Comment by Hejsil** [comment]
This is a better test case:
```rust
test "" {
    const i = failed: {
        // We have to iterate twice
        inline for ([]void{{},{}}) |_| {
            // a, has to be runtime known
            var a = false;

            // We have to have this condition + break twice
            if (a)
                break :failed u8(1);
            if (a)
                break :failed u8(2);
        }

        break :failed u8(0);
    };

    @import("std").debug.assert(i == 0);
}
```

**Comment by Vexu** [comment]
Fixed by 4618c41fa6ca70f06c7e65762d2f38d57b00818c
